rye/src/cli/publish.rs

```diff
@@ -167,6 +194,9 @@ pub fn execute(cmd: Args) -> Result<(), Error> {
     if let Some(cert) = cmd.cert {
         publish_cmd.arg("--cert").arg(cert);
     }
+    if cmd.yes {
+        publish_cmd.arg("--non-interactive");
+    }
```

Comment:
We could also always use `--non-interactive` to avoid prompting with Twine.

```
  --non-interactive     Do not interactively prompt for username/password if
                        the required credentials are missing. (Can also be set
                        via TWINE_NON_INTERACTIVE environment variable.)
```

Reply:
be8424ec2a806cfda96d68da20a66a6d52581100